src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1339,14 +1334,38 @@ private void DecodeInterpolatedStringHandlerArgumentAttribute(ref DecodeWellKnow
                 if (name == "")
                 {
                     // Name refers to the "this" instance parameter.
-                    if (!ContainingSymbol.RequiresInstanceReceiver() || ContainingSymbol is MethodSymbol { MethodKind: MethodKind.Constructor or MethodKind.DelegateInvoke or MethodKind.LambdaMethod })
+                    if (!ContainingSymbol.RequiresInstanceReceiver()
+                        || ContainingSymbol is MethodSymbol { MethodKind: MethodKind.Constructor or MethodKind.DelegateInvoke or MethodKind.LambdaMethod }
+                        || ContainingSymbol.GetIsNewExtensionMember())
                     {
-                        // '{0}' is not an instance method, the receiver cannot be an interpolated string handler argument.
+                        // '{0}' is not an instance method, the receiver or extension receiver parameter cannot be an interpolated string handler argument.
                         diagnostics.Add(ErrorCode.ERR_NotInstanceInvalidInterpolatedStringHandlerArgumentName, arguments.AttributeSyntaxOpt.Location, ContainingSymbol);
                         return null;
                     }
 
-                    return (BoundInterpolatedStringArgumentPlaceholder.InstanceParameter, null);
+                    return BoundInterpolatedStringArgumentPlaceholder.InstanceParameter;
+                }
+
+                if (string.Equals(extensionParameter?.Name, name, StringComparison.Ordinal))
+                {
+                    // We decode the indexes on the symbol that the extension parameter wraps, so to determine equality we need to compare against the underlying parameter.
+                    if (string.Equals(Name, name, StringComparison.Ordinal)
+                        && extensionParameter is ReceiverParameterSymbol { UnderlyingParameter: var underlying }
+                        && (object)underlying == this)
```

Comment:
This feels unnecessary complicated. The marker method is a special method which shouldn't be considered as an extension member. Therefore, I do not think we should be setting ```extensionParameter``` for its parameters. The marker method can be detected by a type check.